SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,257 @@
+
+            
+            _vis3.Brush = _brush1;
+            _vis3.Scale = new Vector3(0, 1, 0);
+            _vis3.Size = new Vector2((float)Rectangle3.ActualWidth, (float)Rectangle3.ActualHeight);
+
+            _vis4.Brush = _brush1;
+            _vis4.Scale = new Vector3(0, 1, 0);
+            _vis4.Size = new Vector2((float)Rectangle4.ActualWidth, (float)Rectangle4.ActualHeight);
+
+            // Parent visuals to XAML rectangles
+            ElementCompositionPreview.SetElementChildVisual(Rectangle1, _vis1);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle2, _vis2);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle3, _vis3);
+            ElementCompositionPreview.SetElementChildVisual(Rectangle4, _vis4);
+
+
+            //create the scale & offset animation
+            Vector3KeyFrameAnimation offsetAnim_r1 = _compositor.CreateVector3KeyFrameAnimation();
+            offsetAnim_r1.InsertKeyFrame(1, new Vector3((float)Rectangle1.ActualWidth, (float)Rectangle1.ActualHeight, 0));
+            offsetAnim_r1.Duration = TimeSpan.FromSeconds(1);
+
+            _scaleAnim = _compositor.CreateVector3KeyFrameAnimation();
+            _scaleAnim.InsertKeyFrame(0, new Vector3(0, 1, 0));
+            _scaleAnim.InsertKeyFrame(.5f, new Vector3(1, 1, 0));
+            _scaleAnim.InsertKeyFrame(1, new Vector3(0, 1, 0));
+            _scaleAnim.Duration = TimeSpan.FromSeconds(2);
+
+
+            // animation of color stops
+            _colorAnimBrush1Stop1 = _compositor.CreateColorKeyFrameAnimation();
+            _colorAnimBrush1Stop1.InsertKeyFrame(.5f, Colors.Honeydew);
+            _colorAnimBrush1Stop1.Duration = TimeSpan.FromSeconds(4);
+
+            _colorAnimBrush1Stop2 = _compositor.CreateColorKeyFrameAnimation();
+            _colorAnimBrush1Stop2.InsertKeyFrame(.5f, Colors.DeepPink);
+            _colorAnimBrush1Stop2.Duration = TimeSpan.FromSeconds(4);
+
+            _offsetAnim = _compositor.CreateScalarKeyFrameAnimation();
+            _offsetAnim.Duration = TimeSpan.FromSeconds(1);
+
+            // when the buttons of text are pressed, the brush will change colors. Below is the set up for animation
+            _changeb1stop1 = _compositor.CreateColorKeyFrameAnimation();
+            _changeb1stop1.InsertKeyFrame(.5f, Colors.LightSkyBlue);
+            _changeb1stop1.Duration = TimeSpan.FromSeconds(2);
+
+            _changeb1stop2 = _compositor.CreateColorKeyFrameAnimation();
+            _changeb1stop2.InsertKeyFrame(.5f, Colors.Teal);
+            _changeb1stop2.Duration = TimeSpan.FromSeconds(2);
+        }
+
+        /*
+         * Run animation on target visual brush
+         */
+
+        private void Pointer_Pressed(object sender, RoutedEventArgs e)
+        {
+            pointerPressedChangeColors(b1GradientStop1, b1GradientStop2);
+        }
+
+        private void pointerPressedChangeColors(CompositionColorGradientStop target, CompositionColorGradientStop target1)
```

Comment:
using a pointerpressed to change gradient color sometimes seems to hit a race condition if you click mid-gradient animation. If there's an easy cleaner way to do this or account for that consider updating (button? bool variable 'lock'). @scott-moore-ms @likuba can maybe recommend?

Reply:
@daneuber can you describe the race condition you're seeing?  I have a guess as to what might be the issue here, but it's related to the fact that she's using target.Color below and PointerPressed is orthogonal...  ie I'd expect that if you click again while an animation is in progress, it restarts the animation rather than reversing it.  Is that what you're seeing?

Reply:
If I click to change the color while over a text item who's gradient animation is currently running one I've seen a few different things happen:
- The gradient changes color mid-animation. Subsequent animated gradients are in the new color scheme
- The gradient does not change color. Subsequent gradients use the new color
- Neither the current or subsequent gradients change color

Reply:
I'm definitely having an issue with this too and keeping it consistent with what I want to happen. trying to troubleshoot it